src/nix/flake.cc

```diff
@@ -1243,25 +1244,62 @@ struct CmdFlakeShow : FlakeCommand, MixJSON
                             j.emplace("description", *description);
                     } else {
-                        logger->cout("%s: %s '%s'",
-                            headerPrefix,
+                        auto type =
                             attrPath.size() == 2 && attrPathS[0] == "devShell" ? "development environment" :
                             attrPath.size() >= 2 && attrPathS[0] == "devShells" ? "development environment" :
                             attrPath.size() == 3 && attrPathS[0] == "checks" ? "derivation" :
                             attrPath.size() >= 1 && attrPathS[0] == "hydraJobs" ? "derivation" :
-                            "package",
-                            name);
+                            "package";
+                        if (description && !description->empty()) {
+                            // Maximum length to print
+                            size_t maxLength = getWindowSize().second > 0 ? getWindowSize().second : 80;
+
+                            // Trim the description and only use the first line
+                            auto trimmed = trim(*description);
+                            auto newLinePos = trimmed.find('\n');
+                            auto length = newLinePos != std::string::npos ? newLinePos : trimmed.length();
+
+                            // Sanitize the description and calculate the two parts of the line
+                            // In order to get the length of the printable characters we need to
+                            // filter out escape sequences.
+                            auto beginningOfLine = fmt("%s: %s '%s'", headerPrefix, type, name);
+                            auto beginningOfLineLength = filterANSIEscapes(beginningOfLine, true).length();
+                            auto restOfLine = fmt(" - '%s'", filterANSIEscapes(trimmed, false, length));
+
+                            // If we are already over the maximum length then do not trim
+                            // and don't print the description (preserves existing behavior)
+                            if (beginningOfLineLength >= maxLength) {
+                                logger->cout("%s", beginningOfLine);
+                            }
+                            else {
+                                auto line = beginningOfLine + restOfLine;
+                                // FIXME: Specifying `true` here gives the correct length
+                                // BUT removes colors/bold so something is not quite right here.
+                                line = filterANSIEscapes(line, true, maxLength);
+
+                                // NOTE: This test might be incorrect since I get things like:
+                                // 168 or 161 > maxLength.
+                                if (line.length() > maxLength) {
+                                    line = line.replace(line.length() - 3, 3, "...");
+                                }
```

Comment:
See the `FIXME:` here. I think this is pretty close but it's not 100% correct yet. `filterANSIEscapes(..., true)` seems to give a reasonable character count but I'm losing color information. `filterANSIEscapes(..., false)` gives funky length values so I think I'm missing something.

Reply:
@edolstra @tomberek it's close! but I'm missing something with the escapes:

![image](https://github.com/user-attachments/assets/00e0b2c8-2b54-4b18-81f9-411a3a29889d)

Reply:
there might be a known number of escapes that you need to correct for, but then also use the original line (plus some terminating ANSI codes). Perhaps do:
1. check length without codes
2. start with string with codes
3. subtract the difference from the max
4. subtract a additional known amount as correction
5. pad with a "reset" ANSI sequence
6. ????
7. Happiness

Reply:
This is all fairly terrible

Reply:
Is there a less perfect solution we could implement for now?

Reply:
It would be nice if `filterANSIEscapes` filtered out color codes.

Reply:
The new version uses a modified version of `filterANSIEscapes` with the tree characters taken into account. It's kind of ugly but seems to work!

Reply:
> This is all fairly terrible.

Absolutely. Terminal color logic, layout logic and flake attribute logic should all be separated out into distinct functions, classes, etc.
However, since we know the solution and we're not dealing with crazy compatibility requirements in this part of the code, I'd be happy to merge this and do the refactoring after.

Reply:
That would be great

Reply:
@roberth Are we good to go as-is?